src/tox/config/loader/ini/replace.py

```diff
@@ -21,28 +22,39 @@
     from tox.config.loader.ini import IniLoader
     from tox.config.main import Config
 
+
+LOGGER = logging.getLogger(__name__)
+
+
 # split alongside :, unless it's preceded by a single capital letter (Windows drive letter in paths)
 ARG_DELIMITER = ":"
 REPLACE_START = "{"
 REPLACE_END = "}"
 BACKSLASH_ESCAPE_CHARS = ["\\", ARG_DELIMITER, REPLACE_START, REPLACE_END, "[", "]"]
+MAX_REPLACE_DEPTH = 100
```

Comment:
Can we add a test for this with a circular reference of length 5, and one with 101?

Reply:
Added a test that checks different recursion levels and asserts that the log message is triggered and that expansion stops at the correct level.